Userland/Applications/ThemeEditor/MainWidget.cpp

```diff
@@ -367,6 +371,80 @@ void MainWidget::save_to_file(Core::File& file)
     }
 }
 
+ErrorOr<Core::AnonymousBuffer> MainWidget::encode()
+{
+    auto buffer = TRY(Core::AnonymousBuffer::create_with_size(sizeof(Gfx::SystemTheme)));
+    auto* data = buffer.data<Gfx::SystemTheme>();
+
+#define __ENUMERATE_ALIGNMENT_ROLE(role) data->alignment[(int)Gfx::AlignmentRole::role] = m_current_palette.alignment(Gfx::AlignmentRole::role);
+    ENUMERATE_ALIGNMENT_ROLES(__ENUMERATE_ALIGNMENT_ROLE)
+#undef __ENUMERATE_ALIGNMENT_ROLE
+
+#define __ENUMERATE_COLOR_ROLE(role) data->color[(int)Gfx::ColorRole::role] = m_current_palette.color(Gfx::ColorRole::role).value();
+    ENUMERATE_COLOR_ROLES(__ENUMERATE_COLOR_ROLE)
+#undef __ENUMERATE_COLOR_ROLE
+
+#define __ENUMERATE_FLAG_ROLE(role) data->flag[(int)Gfx::FlagRole::role] = m_current_palette.flag(Gfx::FlagRole::role);
+    ENUMERATE_FLAG_ROLES(__ENUMERATE_FLAG_ROLE)
+#undef __ENUMERATE_FLAG_ROLE
+
+#define __ENUMERATE_METRIC_ROLE(role) data->metric[(int)Gfx::MetricRole::role] = m_current_palette.metric(Gfx::MetricRole::role);
+    ENUMERATE_METRIC_ROLES(__ENUMERATE_METRIC_ROLE)
+#undef __ENUMERATE_METRIC_ROLE
+
+#define DO_PATH(role, allow_empty)                                                                                                           \
```

Comment:
That's not a very clear name. You can't "do path".

Reply:
Should I also update the name here? https://github.com/SerenityOS/serenity/blob/30b58cd06ccb1855d365d362719cac4f8867c72b/Userland/Libraries/LibGfx/SystemTheme.cpp#L134-L146

Reply:
ha, didn't realize it's stolen code 😄 - gladly!